pwcnet/losses/loss.py

```diff
@@ -50,9 +53,95 @@ def create_multi_level_loss(expected_flow, flows, flow_scaling, flow_layer_loss_
             # Reduce sum in the last 3 dimensions, average over the batch, and apply the weight.
             weight = flow_layer_loss_weights[i]
             layer_loss = weight * tf.reduce_mean(tf.reduce_sum(squared_difference, axis=[1, 2, 3]))
-
-            # Accumulate the total loss.
             layer_losses.append(layer_loss)
-        total_loss += layer_loss
 
+    total_loss = tf.add_n(layer_losses, name='total_loss')
     return total_loss, layer_losses
+
+
+def create_multi_level_unflow_loss(image_a, image_b, forward_flows, backward_flows, flow_scaling,
+                                   flow_layer_loss_weights=None, layer_patch_distances=None, loss_weights=None):
+    """
+    :param image_a: Tensor of shape [B, H, W, 3].
+    :param image_b: Tensor of shape [B, H, W, 3].
+    :param forward_flows: List of tensors of shape [B, H, W, 2]. Flows at different resolutions.
```

Comment:
Mention how these resolutions should be ordered for the default flow_layer_loss_weights somewhere.

Reply:
Done.